src/HttpFactory.php

```diff
@@ -61,7 +61,18 @@ public function createStream(string $content = ''): StreamInterface
      */
     public function createStreamFromFile(string $file, string $mode = 'r'): StreamInterface
     {
-        $resource = \GuzzleHttp\Psr7\try_fopen($file, $mode);
+        try {
+            $resource = \GuzzleHttp\Psr7\try_fopen($file, $mode);
+        } catch (\RuntimeException $e) {
+            // PSR-17 requires to throw an InvalidArgumentException for invalid modes.
+            // But we do not want to validate the modes ourselves as the accepted modes depend on the OS.
+            // The following error messages seem to be returned usually for strange modes.
+            if ('' === $mode || false !== \strpos($e->getMessage(), 'failed to open stream: No error') || false !== \strpos($e->getMessage(), 'failed to open stream: Success')) {
```

Comment:
It seems impossible to conform to PSR-17 here. The error message depends on the OS and on travis it's even wrong: https://travis-ci.org/guzzle/psr7/jobs/542319524
The file tmp file exists but it still says

> Unable to open /tmp/http_factory_tests_AnzmD7 using mode ☠: fopen(/tmp/http_factory_tests_AnzmD7): failed to open stream: No such file or directory

Reply:
So we cannot distinguish between wrong mode or a different failure unless we hardcode the modes.

Reply:
Heh, fun times. This might be close enough. Keep it under 120 though.
```php
if ('' === $mode || false !== \strpos($e->getMessage(), 'failed to open stream: No error')
    || false !== \strpos($e->getMessage(), 'failed to open stream: Success')
) {
```

Reply:
Requires https://github.com/http-interop/http-factory-tests/pull/42

Reply:
Validated mode as per https://github.com/http-interop/http-factory-tests/pull/42#issuecomment-500109322